### ISSUE TITLE
Does it work with S3 ?

### ISSUE THREAD

**Issue by Quiwy** [YunoHost-Apps/restic_ynh#27]
Hi,

I'm revisiting my backup system, which currently uses Duplicity, and I'm considering switching to Restic since it seems to work natively with Yunohost.

The feature description indicates that Restic is compatible with S3, but I’m uncertain about this. During configuration, it prompts for the SFTP port and an SSH user.

From what I’ve read in #23, it seems that S3 support isn’t available at the moment.

Could you clarify if S3 can be used with this app? If not, would it be possible to update the README to avoid any misunderstandings?

Thank you for your help!

**Comment by devloberto** [comment]
I would love using restic_ynh to create and store incremental backups in s3 buckets of my min.io instance. But unfortunately I am pretty sure this app currently does not support s3 at all.

![image](https://github.com/user-attachments/assets/8f74b677-e246-4a50-88f4-3b3420165ba3)

From my point of view it would be good to remove "including S3" from the documentation because it is highly misleading.

**Comment by devloberto** [comment]
I was thinking about alternatively using [backrest](https://github.com/YunoHost-Apps/backrest_ynh/) but likely I'm gonna run [SFTPGo](https://github.com/drakkan/sftpgo) beside the min.io service. This way the restic_ynh app will be connected to SFTPgo which will store the restic repositories in S3 buckets. Maybe such a setup is interesting for you too.

**Comment by devloberto** [comment]
The yunohost online demo confirms my impression of the app's source code:
https://demo.yunohost.org/yunohost/admin/#/apps/install/restic
SFTP and SSH are supported but not S3 at the moment.

**Comment by devloberto** [comment]
Uh I didn't know that SFTP is kind of "FTP over SSH" 😅 sorry for that. So currently only SFTP is supported.

**Comment by devloberto** [comment]
I opened two PRs:

- https://github.com/YunoHost-Apps/restic_ynh/pull/28
- https://github.com/YunoHost/doc/pull/2537
    - https://doc.yunohost.org/en/backup/backup_methods#restic-cf-the-restic-app

**Comment by Gildas-GH** [comment]
Yes, since the last update Restic now works with S3. You need to set the destination URL to your S3 server and add environment variables on the app panel (on YunoHost Admin UI)

**Comment by Quiwy** [comment]
Oh thanks, I'll try soon !

**Comment by devloberto** [comment]
> since the last update Restic now works with S3

Thank you very much! It works like a charm! 🙌🏼
